sky/utils/auth_utils.py

```diff
@@ -58,6 +58,20 @@ def _generate_rsa_key_pair() -> Tuple[str, str]:
     return public_key, private_key
 
 
+def _ensure_key_permissions(private_key_path: str,
+                            public_key_path: str) -> None:
+    """Ensure SSH key files have correct permissions.
+
+    This is necessary because external factors (e.g., Kubernetes fsGroup,
+    volume mounts, umask) can modify file permissions after creation.
+    SSH requires private keys to have strict permissions (0600).
+    """
+    if os.path.exists(private_key_path):
+        os.chmod(private_key_path, 0o600)
+    if os.path.exists(public_key_path):
+        os.chmod(public_key_path, 0o644)
```

Comment:
nit: technically a TOCTTOU race is possible here if the file gets deleted somehow, the alternative would be to catch the `FileNotFoundError`

Reply:
Yes, good catch! Updated it.